packages/typespec-ts/src/utils/credentialUtils.ts

```diff
@@ -0,0 +1,41 @@
+import { NoTarget, Program } from "@typespec/compiler";
+import { Authentication, HttpAuth } from "@typespec/http";
+import { reportDiagnostic } from "../lib.js";
+
+/**
+ * Get supported HTTP authentication schemes and filter out unsupported ones.
+ * @param program used to report diagnostics
+ * @param authorization authentication object
+ * @returns supported HTTP authentication schemes
+ */
+export function getSupportedHttpAuth(
+  program: Program,
+  authorization: Authentication
+): HttpAuth[] {
+  const authList = [];
+  for (const option of authorization.options) {
+    for (const auth of option.schemes) {
+      switch (auth.type) {
+        case "apiKey":
+          if (auth.in === "cookie" || auth.in === "query") {
+            // just skip un-supported authentication and report a warning
+            reportDiagnostic(program, {
```

Comment:
Skip un-supported authentication and report a warning

Reply:
What exactly does cookie mean? I feel like it's just set a header with the name cookie, similar like custom key credential?

Reply:
https://swagger.io/docs/specification/authentication/cookie-authentication/

Simply speaking I don't think cookie auth is just setting a header with customers' input.

Generally browser would automatically maitain the cookie information and this is transparent for end users when trying to communicate with service side. So the server side could save information like session id in Set-Cookie header and when sending any requests these cookies would be sent by Cookie header properly(with scope or some policy) by browser. 

As for node environemnt, if we try to monitor this process we would have a policy which could do something like BearereAuth, 1. go to somewhere to authenticate the users; 2. if successful we will have Set-Cookie header in response. 3. read this info and maintain cookies info in client; 4. append these cookie info to send real requests.